Fix development workflow

Fydon opened the pull request:
@biblemesh/puente-global These changes when used with https://github.com/biblemesh/toad-reader-server/pull/36 result in a fully working development environment when using the commands below. There may be changes to make it better suited for standalone development, but please try running this as a start, particularly for app development. The review process can be used to improve on this, but hopefully this enables development to continue.

Create the external volume for caddy's certificates:
```
docker volume create caddy-data
```

From https://github.com/biblemesh/toad-reader-server/pull/36
```
docker compose --file docker-compose.yml --file docker-compose-override.yml up
```

contents of docker-compose-override.yml below:
```
services:
  caddy:
    image: biblemesh/caddy:latest
    build:
      dockerfile_inline: |
        FROM caddy:builder AS builder

        RUN \
          # nounset - Treat unset variables as an error when substituting
          # xtrace - Print commands and their arguments as they are executed
          set -o nounset -o xtrace && \
          xcaddy build \
            --with github.com/caddyserver/transform-encoder \
            --with github.com/caddyserver/forwardproxy \
            --with github.com/caddyserver/replace-response

        FROM caddy

        COPY --from=builder /usr/bin/caddy /usr/bin/caddy
    pull_policy:
      build

  reader:
    image: biblemesh/toad-reader-server:1.0.0
    pull_policy: build
```

Using toad-reader-apps from this branch, I'm running it with `npm run start` followed by `w` to start the web development.

To extract the root certificate from caddy-data, please use `docker run --rm --user root --volume caddy-data:/backup-volume --volume ./:/backup busybox tar -zcvf /backup/caddy-data.tar.gz /backup-volume`, extract `caddy-data.tar.gz` and obtain it from `backup-volume/caddy/pki/authorities/local/root.crt`. This root certificate will need to be added to your trust store, but is unique to your device.

If you're still experiencing problems, please delete your volumes and some images to ensure that its using the latest setup, e.g. `docker compose down --volumes` and `docker rmi biblemesh/shibboleth-common:1.0.0-dev`.

![chrome_DVcFtxdxt4](https://github.com/user-attachments/assets/4a52ae47-93de-49c0-b677-daca2cd5d41e)

Fydon committed:
Data origin should be https on the default port

Fydon committed:
Use development data origin

Fydon committed:
Add quotes domain

ukch reviewed (approved):
I'm happy with these specific changes. Regarding the whole cross-repo changes you have introduced, these still need to be tested by our team but I'm happy with them in principle.

Fydon commented:
@alitokgonul I'm still seeing Expo builds. Could you please try using the instructions above to run the development environment and report issues if you have any?

Fydon commented:
@ukch I'll wait to hear if you have experienced any issues with using these instructions to run the development environment working before merging this pull request

Fydon commented:
I've deleted `ghcr.io/biblemesh/shibboleth-identity-provider@sha256:b6a33c8e45928d2176935907ae3dc2bdd31316104b70ff589ececb9a9580a3ca` given that its invalid.

I think I had an older `ghcr.io/biblemesh/shibboleth-common:latest` before when `ghcr.io/biblemesh/shibboleth-identity-provider@sha256:e19a243fc0550a9e31d57f2820ca37fb10ba094cc8e26106faf70731e3edab71` was working, so I've pushed a new multi-platform `ghcr.io/biblemesh/shibboleth-identity-provider:latest` using `ghcr.io/biblemesh/shibboleth-common:latest` and updated the `docker-compose-override.yml` above to remove the override for `shibboleth-idp`.

Fydon committed:
Update for npm >= 9
